analytics/upress_reporting/helpers/download_request_parser.py

```diff
@@ -0,0 +1,149 @@
+import os
+import boto3
+import numpy
+import re
+import shutil
+
+from logger import createLog
+from model import Edition, Item, Link
+from model.postgres.item import ITEM_LINKS
+from managers import DBManager
+
+# Regexes needed to parse S3 logs
+REQUEST_REGEX = r"REST.GET.OBJECT "
+# File ID includes the file name for the pdf object
+FILE_ID_REGEX = r"REST.GET.OBJECT (.+pdf\s)"
+TIMESTAMP_REGEX = r"\[.+\]"
+REFERRER_REGEX = r"https://drb-qa.nypl.org/"
```

Comment:
Hmm this referrer regex will be this url for QA but the prod url for production!

Reply:
For now should we keep this in? We could make it an env var once we have the prod URL

Reply:
I was also thinking we could eventually make the publishers env vars (i.e. a list in the yaml configuration)